docs/zpure/index.md

```diff
@@ -0,0 +1,350 @@
+
+We can transform the successful value of a computation using the `map` operator.
+
+```scala mdoc
+val two: ZPure[Nothing, Unit, Unit, Any, Nothing, Int] =
+  one.map(_ + 1)
+```
+
+We can combine the values of two computations with the `zipWith` operator.
+
+```scala mdoc
+val three: ZPure[Nothing, Unit, Unit, Any, Nothing, Int] =
+  one.zipWith(two)(_ + _)
+```
+
+We can also chain computations using the `flatMap` operator and Scala's _for comprehension_ syntax.
+
+```scala mdoc
+val six: ZPure[Nothing, Unit, Unit, Any, Nothing, Int] =
+  for {
+    x <- one
+    y <- two
+    z <- three
+  } yield x + y + z
+```
+
+The easiest way to run a computation is to use the `run` method. It requires us to have already provided any required environment and initial state and handled our errors and just produces the final value.
+
+```scala mdoc
+val value: Int =
+  three.run
+```
+
+## Working With Errors
+
+Since `ZIO` also has an error type the operators for working with errors are quite similar to the ones on `ZIO`.
+
+We can create a computation that fails with an error using the `fail` operator.
+
+```scala mdoc
+val fail: ZPure[Nothing, Unit, Unit, Any, String, Nothing] =
+  ZPure.fail("fail")
+```
+
+Notice how now the error type is `String`, indicating that this computation can fail with a `String` error. In this case the success type is `Nothing` since we actually know that this computation can never succeed.
+
+If we are working with code that can throw exceptions we can use the `attempt` operator of `ZPure` to safely import that code into our computation, catching any non-fatal exceptions and translating them into failure values.
+
+```scala mdoc
+def parseIntThrowable(s: String): ZPure[Nothing, Unit, Unit, Any, Throwable, Int] =
+  ZPure.attempt(s.toInt)
+```
+
+We can use the `mapError` operator to transform the error type of a computation, for example to map it to a common error type we are using in other parts of our computation.
+
+```scala mdoc
+def parseInt(s: String): ZPure[Nothing, Unit, Unit, Any, String, Int] =
+  parseIntThrowable(s).mapError(_.getMessage)
+```
+
+We can recover from errors by using the `catchAll` operator, which allows us to recover from the failure using a new computation.
+
+```scala mdoc
+def parseIntOrDefault(s: String): ZPure[Nothing, Unit, Unit, Any, Nothing, Int] =
+  parseInt(s).catchAll(_ => ZPure.succeed(0))
+```
+
+Now the error type is `Nothing`, indicating that we have handled all our errors.
+
+There are many other error handling operators defined on `ZPure`. The `catchSome` operator allows us to only recover from certain errors.
+
+The `fold` and `foldM` operators allow us to handle both the failure and success cases at the same time. And the `orElse` operator allows us to specify a fallback computation that will be run if the original computation fails.
+
+Since these operators are the same as the ones on `ZIO` we will not cover them all in detail here.
```

Comment:
I'd be inclined to make this statement sooner by stating something along: we'll discuss some of the important operators of ZPure that have correspondence and good documentation in ZIO but will not give all details here such as method signatures as the reader can obtain them directly in Scala docs or assume familiarity with ZIO. The list of operators include: catchSome, mapError, fold, foldM, ...

My point is here that as I read along I am thinking why don't I see the operator type signatures just like I did elsewhere in docs for Prelude? And the explanation comes in later.